osu.Game/Rulesets/Filter/FilterCriteria_QueryParser.cs

```diff
@@ -93,120 +139,122 @@ private static void parseKeywordCriteria(FilterCriteria criteria, string key, st
         private static bool parseInt(string value, out int result) =>
             int.TryParse(value, NumberStyles.None, CultureInfo.InvariantCulture, out result);
 
-        private static void updateCriteriaText(ref FilterCriteria.OptionalTextFilter textFilter, string op, string value)
+        private static bool updateCriteriaText(ref OptionalTextFilter textFilter, Operator op, string value)
```

Comment:
These helpers maybe should be `protected` too, not sure. If they are to be `protected`, then could be an argument for splitting parsing out to a separate class after all.